cmd/mdltool/main.go

```diff
@@ -176,10 +183,21 @@ func cmdPackage(args []string) int {
 		fmt.Fprintf(os.Stderr, "Continuing anyway, but this may cause issues.\n")
 	}
 
-	// Parse the reference
-	target, err := registry.NewClient(
+	// Create registry client with auth if environment variables are set
+	registryClient := registry.NewClient(
```

Comment:
Minor consideration: if `registry.NewClient` starts becoming more expensive, then it might be wiser to define `var registryClient *registry.Client` and then construct / assign to it with an `else`.  Probably okay for now since cheap.